internal/boot/helpers.go

```diff
@@ -63,3 +64,16 @@ func GenerateRandomString(prefix string) (string, error) {
 
 	return prefix + id.String(), nil
 }
+
+// GenerateCIArtifactName generates a new identifier for CI artifacts which is based
+// on environment variables specified by Jenkins
+// note: in case of migration to sth else like Github Actions, change it to whatever variables GH Action provides
+func GenerateCIArtifactName(prefix string) (string, error) {
+	changeId := os.Getenv("CHANGE_ID")
+	buildId := os.Getenv("BUILD_ID")
```

Comment:
I don't like reading environment variables from a library, because it makes it hard to figure out how a program is meant to be run. Is there a way to pass those through from `main_test.go`?

Don't block on it, especially since there seems to be a lot of such use here, and it's only test code.

Reply:
We could move the whole function out of the library into `main_test.go`, the only function which requires separate solutions is `WithBootedImageInEC2` because it also creates a security group, but we can pass the change/build id to it.